spec/integration/aws_internet_gateway_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe Chef::Resource::AwsInternetGateway do
+  extend AWSSupport
+
+  when_the_chef_12_server "exists", organization: 'foo', server_scope: :context do
+    with_aws "with a VPC and an internet gateway" do
+      vpc = nil
+      internet_gateway = nil
+
+      before {
```

Comment:
I've found that with that approach if an unexpected error happens which causes rspec to fail - the vpc (or really any other supporting resource created so far) wouldn't get destroyed - hence the "manual" before/after approach.

I understand the value/easiness of reusing chef resources in the specs but personally I'd like to avoid having to manually check/destroy test resources in case of unexpected failure of the test framework.

I'll change it if you think it'd better anyway - let me know.